src/Pos/Util/UserSecret.hs

```diff
@@ -134,7 +188,19 @@ writeUserSecretRelease u
 
 -- | Helper for writing secret to file
 writeRaw :: UserSecret -> IO ()
-writeRaw u = BSL.writeFile (u ^. usPath) $ encode u
+writeRaw u = do
+    let path = u ^. usPath
+    -- On POSIX platforms, openTempFile guarantees that the file
+    -- will be created with mode 600.
+    bracket
+        (openBinaryTempFile (takeDirectory path) (takeFileName path))
+        (\(tempPath, tempHandle) -> do
```

Comment:
So you want the file to be created even if `BSL.hPut tempHandle $ encode u` fails?